lib/dpul_collections_web/live/search_live.ex

```diff
@@ -3,47 +3,90 @@ defmodule DpulCollectionsWeb.SearchLive do
   alias DpulCollections.Solr
 
   defmodule Item do
-    defstruct [:id, :title]
+    defstruct [:id, :title, :date]
   end
 
   def mount(_params, _session, socket) do
     {:ok, socket}
   end
 
   def handle_params(params, _uri, socket) do
-    filter = %{
-      q: valid_query(params)
+    filters = %{
```

Comment:
Okay I'm thinking this is the solr search. Maybe we could make a module called `Solr.Search` and have a struct for these fields and then the initialize logic and anything else that's specific to the search itself.

Reply:
I not really in favor of abstracting them away at the moment. They are indeed used to generate solr_params, but are not the solr params themselves. They are the live view parameters and are used in various places in the template. It's useful, IMO, to have a ready access to the params along with the code and markup at this stage. We're in heavy development and are all learning this new framework.

Reply:
Okay. Could we think about naming it something else within this file? It is more than the search filters, since it's got the query as well. It's like a piece of the solr parameter pipeline in between the params that come from the form and the params we send to solr.

Reply:
Created a nested SearchState module to handle this. The `from_params` method returns a map instead of struct because it's much easier right now to handle maps. We pass these as url params ([example](https://github.com/pulibrary/dpul-collections/pull/149/files#diff-e1587d7e7279a676adfb0beaddc470fb9342beec6be8898730a652ed01a61bb9R174)) and this doesn't work well with a struct. Might be able to do something to change this in the future if it makes sense.